### PR TITLE
Desert Course - Ice-Cream Edition. Ports Frozen based deserts from TG.

### PR DESCRIPTION
**Pull request by Mitchs98**
## What Does This PR Do
Ports 16 snowcones, 10 popsicles, and 8 frozen treats from TG. Creates a new frozen.dmi and frozen.dm file to house all the frozen goodies within, also moves ice-cream vat ice-cream code and icons to those new files as well as the ice-cream sandwich. 

Also adds Popsicle Sticks, for popsicle related reason. And a recipe for Waffle Cones for ice-cream related reasons.

I will look for PR credits to try and properly credit TG spriters directly tomorrow.

Recipes;

<details>
<summary>Snowcones</summary>
Apple Snowcone - 1 paper cup + 15u ice + 5u Apple Juice

Berry Snowcone - 1 paper cup + 15u ice + 5u Berry Juice

Bluecherry Snowcone - 1 paper cup + 15u ice + 5u Bluecherry Jelly

Cherry Snowcone - 1 paper cup + 15u ice + 5u Cherry Jelly

Flavorless Snowcone - 1 paper cup + 15u ice

Fruit Salad Snowcone - 1 paper cup + 15u ice + 5u Orange Juice + 5u Lemon Juice + 5u Lime Juice

Grape Snowcone - 1 paper cup + 15u ice + 5u Grape Juice

Honey Snowcone - 1 paper cup + 15u ice + 5u Honey

Lemon Snowcone - 1 paper cup + 15u ice + 5u Honey Juice

Lime Snowcone - 1 paper cup + 15u ice + 5u Lime Juice

Mime Snowcone - 1 paper cup + 15u ice + 5u Nothing

Orange Snowcone - 1 paper cup + 15u ice + 5u Orange Juice

Pineapple Snowcone - 1 paper cup + 15u ice + 5u Pineapple Juice

Rainbow Snowcone - 1 paper cup + 15u ice + 5u Colorful Reagent

Space Cola Snowcone - 1 paper cup + 15u ice + 5u Cola

Space Mountain Wind Snowcone - 1 paper cup + 15u ice + 5u Space Mountain Wind
</details>

<details>
<summary>Popsicles</summary>
Banana Topsicle - 1 tofu + 5u Banana Juice + 5u Sugar + 1 Popsicle Stick

Berry Topsicle - 1 tofu + 5u Berry Juice + 5u Sugar + 1 Popsicle Stick

Pineapple Topsicle - 1 tofu + 5u Pineapple Juice + 5u Sugar + 1 Popsicle Stick

Berry Creamsicle - 1 Popsicle Stick, 4u Berry Juice, 2u Ice, 2u Milk Cream, 2u Sugar, 2u Vanilla

Licorice Creamsicle - 1 Popsicle Stick, 4u Blumpkin Juice, 2u Ice, 2u Milk Cream, 2u Sugar, 2u Vanilla

Orange Creamsicle - 1 Popsicle Stick, 4u Orange Juice, 2u Ice, 2u Milk Cream, 2u Sugar, 2u Vanilla

Sea-Salt Ice-Cream Bar - 5u cream, 5u sugar, 3u Salt, 1 Popsicle Stick

Jumbo Ice-Cream - 1 Popsicle Stick, 1 Chocolate Bar, 2u Ice, 2u Milk Cream, 2u Sugar, 3u Vanilla

Frozen Pineapple Pop - 1 Pineapple Slice, 1 Chocolate Bar, 1 Popsicle Stick

Ant Popsicle - 1 Popsicle Stick, 5u Sugar, 5u Water, 10u Ants
</details>

<details>
<summary>Frozen Treats</summary>
Cornuto - 1 Chocolate Bar, 1 Flat Dough, 2u Ice, 4u Milk Cream, 4u Sugar

Honkdae - 5u Cream, 1 Clown Mask, 1 Cherry, 2 Banana, 1 Icecream

Peanut Butter Ice Cream Mochi - 5u Sugar, 5u Rice, 2u Peanut Butter, 1 Waffle Cone

Space Freezy - 5u Blue Cherry Jerry, 15u Space Mountain Wind, 1 Icecream

Strawberry Ice-Cream Sandwich - 2 Berries, 5u Ice, 5u milk cream, 1 Waffle Cone

Sundae - 5u Cream, 1 Cherry, 1 Banana, 1 Waffle Cone
</details>

## Images of changes
dreamseeker_B8LEEN8d7U

dreamseeker_IuWLRYRn5d

dreamseeker_1qO71EN9IM

## Testing
1. Spawned 1 of everything.
2. Made 1 of everything.

## Changelog
:cl:
add: Adds 16 snowcones, 10 popsicles, and 8 frozen treats from TG.
/:cl:

### CODE REVIEW

**Comment by Henri215** [comment]
image
All of this is not necessary on the changelog, if you are adding a new item its already expected for it to come with a sprite and the others are not player facing changes.

**Comment by Mitchs98** [comment]
> image All of this is not necessary on the changelog, if you are adding a new item its already expected for it to come with a sprite and the others are not player facing changes.

Alright. Had similar stuff in the previous CL, but I'll edit it.

File: code/modules/food_and_drinks/food/foods/frozen.dm [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+	name = "jumbo icecream"
+	desc = "A luxurious ice cream covered in rich chocolate. It seems smaller than you remember it being."
+	icon_state = "jumbo"
+	trash = /obj/item/trash/popsicle_stick
+	list_reagents = list("nutriment" = 4, "sugar" = 4, "chocolate" = 3)
+	tastes = list("ice cream" = 1, "chocolate" = 1, "vanilla" = 1)
+
+/obj/item/reagent_containers/food/snacks/frozen/popsicle/bananatop
+	name = "banana topsicle"
+	desc = "A frozen treat made from tofu and banana juice blended smooth, then frozen. Popular in rural Japan in the summer."
```
Comment (Henri215) [inline code/modules/food_and_drinks/food/foods/frozen.dm:276]: Japan still exists? I have no idea :D

File: code/modules/food_and_drinks/kitchen_machinery/processor.dm [hunk 2]
```diff
@@ -119,6 +119,10 @@
 	input =  /obj/item/reagent_containers/food/snacks/grown/carrot
 	output = /obj/item/reagent_containers/food/snacks/grown/carrot/wedges
 
+/datum/food_processor_process/towercap
+	input =  /obj/item/grown/log
```
Comment (Henri215) [inline code/modules/food_and_drinks/kitchen_machinery/processor.dm:123]:
```suggestion
	input = /obj/item/grown/log
```

File: code/game/objects/items/trash.dm [hunk 3]
```diff
@@ -126,5 +126,10 @@
 	desc = "Not sticky anymore."
 	throw_range = 1
 
+/obj/item/trash/popsicle_stick
+	name = "used popsicle stick"
+	icon_state = "popsicle_stick_s"
+	desc = "Still tastes sweet."
```
[the remark this answers is not in the document]
Comment (Mitchs98) [reply code/game/objects/items/trash.dm:132]: Have you ever licked an old popsicle stick?

File: code/modules/food_and_drinks/food/foods/frozen.dm [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+	icon = 'icons/obj/food/frozen_treats.dmi'
+	icon_state = "clowncake"
+	slice_path = /obj/item/reagent_containers/food/snacks/clowncakeslice
+	list_reagents = list("nutriment" = 20, "sugar" = 5, "vitamin" = 5, "banana" = 15)
+	tastes = list("cake" = 5, "sweetness" = 2, "carrot" = 1)
+
+/obj/item/reagent_containers/food/snacks/clowncakeslice
+	name = "clown cake slice"
+	desc = "A slice of bad jokes, and silly props."
+	icon = 'icons/obj/food/frozen_treats.dmi'
```
[the remark this answers is not in the document]
Comment (Mitchs98) [reply code/modules/food_and_drinks/food/foods/frozen.dm:26]: Didn't think about that for the slice.

**Comment by Oyu07** [comment]
279482255-0db288c3-ea15-445f-adbb-6e7f3ec28d80
Why the long gap? just curious

**Comment by Mitchs98** [comment]
> 279482255-0db288c3-ea15-445f-adbb-6e7f3ec28d80 Why the long gap? just curious

Gonna be real.

Was too lazy to make it not have the gap. I'll change it tomorrow.

File: code/modules/food_and_drinks/food/foods/frozen.dm [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+	tastes = list("cake" = 5, "sweetness" = 2, "carrot" = 1)
+
+/obj/item/reagent_containers/food/snacks/frozen/clowncakeslice
+	name = "clown cake slice"
+	desc = "A slice of bad jokes, and silly props."
+	icon_state = "clowncake_slice"
+	trash = /obj/item/trash/plate
+	filling_color = "#FFD675"
+	list_reagents = list("nutriment" = 4, "sugar" = 1, "vitamin" = 1, "banana" = 3)
+	tastes = list("cake" = 5, "sweetness" = 2, "sad clowns" = 1, "ice-cream")
```
Comment (Sirryan2002) [inline code/modules/food_and_drinks/food/foods/frozen.dm:30]:
```suggestion
	tastes = list("cake" = 5, "sweetness" = 2, "sad clowns" = 1, "ice-cream" = 1)
```

**Review by Sirryan2002 — APPROVED** [review]

**Review by S34NW — APPROVED** [review]

**Pull request merged**